ui-core/src/maintenance/MaintenanceMode.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useState } from 'react'
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faScrewdriverWrench } from '@fortawesome/free-solid-svg-icons'
+import { Markdown, useApiContext } from '@juniper/ui-core'
+import { MaintenanceModeSettings } from 'src/types/maintenance'
+
+export function MaintenanceMode({ children }: { children: React.ReactNode }) {
+  const [settings, setSettings] = useState<MaintenanceModeSettings>()
+  const [bypassMaintenanceMode, setBypassMaintenanceMode] = useState(false)
+  const [password, setPassword] = useState('')
+  const Api = useApiContext()
+
+  useEffect(() => {
+    const loadSettings = async () => {
```

Comment:
we may want to get slightly more sophisticated here and periodically poll the maintenance mode endpoint in case users are actively using the system when we enter maintenance mode. as it stands right now, only a hard-refresh would reload the latest settings